sdk/nodejs/admissionregistration/v1beta1/MutatingWebhookConfiguration.ts

```diff
@@ -77,7 +78,23 @@ import { getVersion } from "../../version";
        * @param opts A bag of options that control this resource's behavior.
        */
       constructor(name: string, args?: inputApi.admissionregistration.v1beta1.MutatingWebhookConfiguration, opts?: pulumi.CustomResourceOptions) {
+          // In Kubernetes, `.metadata.namespace` must be a `string`. In Pulumi, we allow users to
+          // pass in `string | core.v1.Namespace`. Thus, if the user has provided a
+          // `core.v1.Namespace`, we convert it to a `string` representing that namespace's name.
```

Comment:
I think that you can avoid this code iff the a `Namespace`'s `id` is its name: the RPC layer marshals a `Resource` instance as the value of its `id` field.

Reply:
We could do this, but in the longer term I wasn't sure whether this would be the `id` ([_cf_](https://github.com/pulumi/pulumi-kubernetes/issues/353).), and in any event, I personally value the explicitness of encoding relationship between resource as code vs. using parts of the Pulumi protocol that happen to contain that info, just to shorten the code. My 2 cents.

Reply:
> I personally value the explicitness of encoding relationship between resource as code vs. using parts of the Pulumi protocol that happen to contain that info, just to shorten the code. My 2 cents.

That's fair. I don't think of this as a nice coincidence, though--the fact that we marshal resources as IDs is very intentional, and is designed to support exactly this sort of scenario without the boilerplate.